mapper/pojo-standalone/src/main/java/org/hibernate/search/mapper/pojo/standalone/mapping/SearchMappingBuilder.java

```diff
@@ -95,11 +96,55 @@ public SearchMappingBuilder properties(Map<String, ?> map) {
 		return this;
 	}
 
+
+	/**
+	 * Reads the properties from the reader and sets them as overrides for already configured properties.
+	 * <p>
+	 * Provided reader should be compatible with {@link java.util.Properties#load(Reader)}.
+	 * <p>
+	 * Configuration properties are mentioned in {@link org.hibernate.search.mapper.pojo.standalone.cfg.StandalonePojoMapperSettings},
+	 * or in the reference documentation for backend-related properties.
+	 *
+	 * @param propertiesReader A configuration property source reader.
+	 * Properties from it will be added as an override to previously set properties.
+	 * @return {@code this}, for call chaining.
+	 * @see ConfigurationPropertySource#withOverride(ConfigurationPropertySource)
+	 */
+	public SearchMappingBuilder properties(Reader propertiesReader) throws IOException {
+		// "Reset" the map so the following properties added to it will not be potentially overridden
+		// by a properties we are about to read form the reader.
+		this.propertySource = properties.isEmpty()
+				? ConfigurationPropertySource.empty()
+				: getPropertySource( properties, propertyChecker );
+		this.properties = new HashMap<>();
+		this.propertySource = this.propertySource
+				.withOverride(
+						propertyChecker.wrap( AllAwareConfigurationPropertySource.fromPropertiesReader( propertiesReader ) ) );
+
+		return this;
+	}
+
 	/**
 	 * Builds the search mapping.
 	 * @return The {@link SearchMapping}.
 	 */
 	public CloseableSearchMapping build() {
+		// Double-check if we've added any properties to the map before building the mapping.
+		// We do so to make sure we are tracking all the property keys:
+		if ( propertySource != null ) {
+			if ( !properties.isEmpty() ) {
+				// means we already had a property source (read a file) and we've added some properties after that too,
+				// but those properties in the map are not yet tracked or converted to a property source:
+				propertySource = propertySource.withOverride( getPropertySource( properties, propertyChecker ) );
+			}
+		}
+		else {
+			// means we were only adding properties to the map, so we need to create a source before moving forward:
+			propertySource = properties.isEmpty()
+					? ConfigurationPropertySource.empty()
+					: getPropertySource( properties, propertyChecker );
+		}
+
```

Comment:
It seems like this would get much simpler (and robust?) if upon calling `.properties()` you just put the content of the file into `this.properties`... ?

Reply:
IDK 😃, the only reason for keeping "multiple" property sources would be to be able to separate the properties and know where they came from. But if there's no real benefit in that then we can simplify it. Let me know how to proceed 😃

Reply:
That would be nice, but would also require that we somehow keep the name of the source file in the `ConfigurationPropertySource` for debugging purposes. And since the input is a `Reader`... we can't do that.

I'd favor simplifying this, then?

Reply:
From a new configuration property source to a single couple-lines-method 😄

Reply:
Yeah, I suppose that's how experimentation works: sometimes you have to put in effort to determine what you should *not* do :) Thanks for taking care of this!